### ISSUE TITLE
Problems with virtualenvs

### ISSUE THREAD

**Issue by cjauvin** [proofit404/anaconda-mode#281]
I started using`anaconda-mode` recently and I quickly ran into problems trying to make it work with virtualenvs on Linux (using `pythonic-activate`): it doesn't seem to be able to find definitions for packages that were pip installed in a specific virtualenv. I investigated the problem and came up with a solution using `jedi.create_environment`:

http://jedi.readthedocs.io/en/latest/docs/api.html#jedi.create_environment

for which I will propose a PR shortly.

**Comment by proofit404** [comment]
Hm...

The initial design which works solid good for 4+ years was to run an anaconda-mode server from within the virtual environment, so packages in it are visible to Jedi.

Does this thing change recently?

**Comment by cjauvin** [comment]
My understanding of the design is not very deep (I basically learned everything I know about `jedi` today), so I might be doing something silly, but there is clearly something wrong on my laptop, and when I tried to dig into it, I realized that doing something like this

```shell
/path/to/venv/bin/python script_using_jedi_goto_definitions.py`
```

will not work with symbols from packages available only in that venv. I found only two ways to make it work:

1. Set VIRTUAL_ENV (to `/path/to/venv`)
or
2. Use `jedi.create_environment`

I read in jedi's doc that virtualenv support has been quite problematic for quite a while, and that (2) was a new mechanism to help, so that's why I investigated it and proposed a PR making use of it.
